### PR TITLE
[feature] Drag & drop for create/edit questions forms

### PR DESCRIPTION
**Pull request by galer7**
## Changes
- Implement drag and drop with drag overlay ([docs guide](https://docs.dndkit.com/presets/sortable#drag-overlay))
- The effect on the questions array is that of "place this element here and move the rest below it", not that of a replace/switch

## Further (possible) improvements:
- [x] The changes from `src/components/QuestionListEdit/QuestionListEdit.tsx` can probably be placed in a new context for a tidier implementation (`QuestionsDndContext`)
- [ ] We can provide some padding to the questions in order to have a place from which we can drag
- [ ] We can have a line element that showcases where the question that is being dragged will be placed

### CODE REVIEW

File: src/components/QuestionListEdit/QuestionDndContext.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      <DragOverlay>
+        {activeQuestion ? (
+          <QuestionDragOverlay question={activeQuestion} />
+        ) : null}
+      </DragOverlay>
+    </DndContext>
+  );
+};
+
+const QuestionDragOverlay: React.FC<{ question: string }> = forwardRef(
```
Comment (razvan-thinslices) [inline src/components/QuestionListEdit/QuestionDndContext.tsx:76]: Can we use the QuestionForm component in order for this to have the same aspect as the component you're actually moving?

File: src/utils/sensors.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  ];
+  if (
+    element?.tagName &&
+    interactiveElements.includes(element.tagName.toLowerCase())
+  ) {
+    return true;
+  }
+
+  return false;
+}
```
Comment (razvan-thinslices) [inline src/utils/sensors.ts:43]: Can we create a component that will act as a handle for drag and drop instead of having the whole question draggable?

**Review by razvan-thinslices — COMMENTED** [review]
Like i mentioned in one of the comments, I would create a component to act as a handle to initiate the drag and drog process.

>We can have a line element that showcases where the question that is being dragged will be placed

Do you intent on implementing this in this pull request?

File: src/utils/sensors.ts (hunk 2 above)
Comment (galer7) [reply src/utils/sensors.ts:43]: Do you mean like some sort of anchor/spot from which we would drag the question?

Comment (razvan-thinslices) [reply src/utils/sensors.ts:43]: Yes

**Pull request closed without merging**